File: src/main/java/chloe/movietalk/common/RedisUtil.java
package chloe.movietalk.common;

import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

import java.time.Duration;
import java.util.concurrent.TimeUnit;

@Component
@RequiredArgsConstructor
public class RedisUtil {
    private final RedisTemplate<String, Object> redisTemplate;

    public void setBlacklist(String key, Object o, Duration minutes) {
        redisTemplate.opsForValue().set(key, o, minutes);
    }

    public Object getBlacklist(String key) {
        return redisTemplate.opsForValue().get(key);
    }

    @Transactional
    public void setValuesWithTimeout(String key, String value, Long timeout) {
        redisTemplate.opsForValue().set(key, value, timeout, TimeUnit.MILLISECONDS);
    }

    @Transactional
    public void deleteValues(String key) {
        redisTemplate.delete(key);
    }
}

File: src/main/java/chloe/movietalk/common/RedisUtil.kt
package chloe.movietalk.common

import lombok.RequiredArgsConstructor
import org.springframework.data.redis.core.RedisTemplate
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional
import java.time.Duration
import java.util.concurrent.TimeUnit

@Component
@RequiredArgsConstructor
class RedisUtil {
    private val redisTemplate: RedisTemplate<String?, Any?>? = null

    fun setBlacklist(key: String, o: Any, minutes: Duration) {
        redisTemplate!!.opsForValue().set(key, o, minutes)
    }

    fun getBlacklist(key: String): Any? {
        return redisTemplate!!.opsForValue().get(key)
    }

    @Transactional
    fun setValuesWithTimeout(key: String, value: String, timeout: Long) {
        redisTemplate!!.opsForValue().set(key, value, timeout, TimeUnit.MILLISECONDS)
    }

    @Transactional
    fun deleteValues(key: String) {
        redisTemplate!!.delete(key)
    }
}
